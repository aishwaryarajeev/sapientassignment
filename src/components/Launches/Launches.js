import React, { useEffect, useState } from "react";
import Launch from "../Launch";
import { makeStyles } from "@material-ui/styles";

const useStyles = makeStyles({
  loader: {
    display: "flex",
    justifyContent: "center",
  },
  root: {
    width: "fit-content",
    margin: "10px",
  },
  container: {
    width: "100%",
    display: "flex",
    flexWrap: "wrap",
    justifyContent: "center",
  },
  img: {
    backgroundColor: "#e6e6e6",
  },
  missionID: {
    fontWeight: 600,
    fontSize: "14px",
  },
  values: {
    margin: 0,
    fontSize: "13px",
    color: "#002b80",
  },
  heading: {
    marginBottom: "12px",
    fontSize: "12px",
    fontWeight: 600,
    color: "#002b80",
  },
});

export const Launches = ({ endPoint }) => {
  const [initialData, setInitialData] = useState([]);
  const classes = useStyles();

  useEffect(() => {
    fetch(endPoint)
      .then((res) => res.json())
      .then((data) => {
        setInitialData(data);
      })
      .catch(console.log);
  }, [endPoint]);
  return (
    initialData.length > 0 && (
      <div className={classes.container}>
        {initialData.map((data) => {
          return (
            <Launch
              data-testid="Launch"
              key={data.flight_number}
              mission={data}
            />
          );
        })}
      </div>
    )
  );
};
