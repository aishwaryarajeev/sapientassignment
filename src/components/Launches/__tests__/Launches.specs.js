import React from "react";
import { shallow, mount } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import { Launches } from "../Launches";

describe("<Launches />", () => {
  let wrapper = null;
  beforeEach(() => {
    wrapper = shallow(<Launches />);
  });
  it("should render correctly", () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it("should render correctly", () => {
    wrapper = mount(<Launches />);
    wrapper.update();

    expect(wrapper.find("Launch")).toBeTruthy();
  });
});
