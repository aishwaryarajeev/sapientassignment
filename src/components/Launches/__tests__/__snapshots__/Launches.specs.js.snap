// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Launches /> should render correctly 1`] = `""`;
